compiler/rustc_passes/src/dead.rs

```diff
@@ -830,14 +830,16 @@ fn check_mod_deathness(tcx: TyCtxt<'_>, module: LocalDefId) {
     for item in module_items.items() {
         if !live_symbols.contains(&item.owner_id.def_id) {
             let parent = tcx.local_parent(item.owner_id.def_id);
-            if parent != module && !live_symbols.contains(&parent) {
+            if parent != module
+                && !live_symbols.contains(&parent)
+                && !matches!(tcx.def_kind(parent), DefKind::Closure | DefKind::Generator)
```

Comment:
Maybe we should just mark closures as live when walking the HIR tree?

Either way, this needs an explanation comment.

Reply:
Moved this to `check_item` to consider it live in the initial walk